### PR TITLE
M3-3453 Update domainsDrawer reducer to use similar pattern as volumesDrawer reducer

### PR DESCRIPTION
**Pull request by WilkinsKa1**
## Description

domainsDrawer was using an old `actionCreator` function that wasn't type-safe or consistent with the pattern of volumesDrawer. This updates domainsDrawer to utilize `actionCreatorFactory` instead of the old function.

## Type of Change
- Non breaking change ('update')

## Note to Reviewers

Please make sure there aren't any regressions to the domains drawer and that the `Create Domain` analytics are working as expected.

### CODE REVIEW

File: packages/manager/src/store/domainDrawer/index.ts [hunk 1]
```diff
@@ -36,39 +38,43 @@ interface Close extends Action {
...
-export const openForCreating: ActionCreator = (origin: Origin): Creating => ({
-  type: CREATING,
-  origin
+export const openForCreating = (origin: Origin) => createDomain({ origin });
+
+interface CreateDomainPayload {
+  origin: Origin;
+}
+
+const createDomain = actionCreator<CreateDomainPayload>(`CREAT_DOMAIN`, {
```
Comment (martinmckenna) [inline packages/manager/src/store/domainDrawer/index.ts]: misspelling

Comment (WilkinsKa1) [reply packages/manager/src/store/domainDrawer/index.ts]: good catch, thanks!

File: packages/manager/src/store/domainDrawer/index.ts [hunk 2]
```diff
@@ -16,9 +9,18 @@ export interface State {
   origin?: Origin;
 }
 
+const actionCreator = actionCreatorFactory(`@@manager/domains`);
+
+// ACTIONS
+export const OPEN = '@manager/domains/OPEN';
+export const CLOSE = '@manager/domains/CLOSE';
```
Comment (Jskobos) [inline packages/manager/src/store/domainDrawer/index.ts]: can we rename these to `@manager/domainDrawer/WHATEVER`? It's hard to follow this way in devtools since there's two types of Domain state in Redux

Comment (WilkinsKa1) [reply packages/manager/src/store/domainDrawer/index.ts]: updated!
